src/FileDownloader.ts

```diff
@@ -119,6 +120,20 @@ export default class FileDownloader implements IFileDownloader {
             throw new DownloadCanceledError();
         }
 
+        // Make the file executable if requested
+        // File permissions are preserved after moving the file
+        try {
+            if (makeExecutable) {
+                await fs.promises.chmod(tempFileDownloadPath, 0o700);
```

Comment:
💡 thank you so much for this - just a note please 📝 here - do we know what was the default behaviour? With this permission it will have no permission left for user and group users so just fyi to make sure we documented what was the default behaviour vs this.

Reply:
Great catch - I checked the permissions when it is downloaded, put a comment in the code and we are now using 744 (originally 644)